src/jit/valuenum.cpp

```diff
@@ -6892,6 +6903,15 @@ void Compiler::fgValueNumberTree(GenTreePtr tree, bool evalAsgLhsInd)
                             vnStore->VNExcSetSingleton(vnStore->VNForFunc(TYP_REF, VNF_OverflowExc));
                         excSet = vnStore->VNPExcSetUnion(excSet, ValueNumPair(overflowExcSet, overflowExcSet));
                     }
+
+                    // Divides add a divide-by-zero exception.
+                    if (tree->OperIs(GT_DIV, GT_UDIV))
+                    {
+                        ValueNum divideByZeroExcSet =
+                            vnStore->VNExcSetSingleton(vnStore->VNForFunc(TYP_REF, VNF_DivideByZeroExc));
```

Comment:
`VNF_DivideByZeroExc` needs to be updated to take the divisor as an argument for the same reasons that `VNF_NullPtrExc` takes the pointer as an argument.

It would be good to add a testcase that fails without doing so.
For that matter, it would be good to add the repro cases from #8648.

`VNF_OverflowExc` ought to take the value number for the potentially-overflowing operation as an argument, too, but that could be a separate issue/change.